classes/class-p4-metabox-register.php

```diff
@@ -0,0 +1,625 @@
+			'Shifting the trillions'             => 'Shifting the trillions',
+			'Stolen Fish'                        => 'Stolen Fish',
+			'Amazon Reef'                        => 'Amazon Reef',
+			'People vs. Oil'                     => 'People vs. Oil',
+			'Pipelines'                          => 'Pipelines',
+			'Ends of the Earth'                  => 'Ends of the Earth',
+			'Patagonia'                          => 'Patagonia',
+			'Urban Revolution'                   => 'Urban Revolution',
+		];
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Campaign Name', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'campaign_name',
+				'type'       => 'select',
+				'options'    => $campaign_options,
+				'attributes' => [
+					'data-validation' => 'required',
+				],
+			]
+		);
+
+		$basket_options = [
+			'not set'              => __( '- Select Basket -', 'planet4-master-theme-backend' ),
+			'Forests'              => 'Forests',
+			'Oceans'               => 'Oceans',
+			'Good Life'            => 'Good Life',
+			'Food'                 => 'Food',
+			'Climate & Energy'     => 'Climate & Energy',
+			'Oil'                  => 'Oil',
+			'Political & Business' => 'Political & Business',
+		];
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Basket Name', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'basket_name',
+				'type'       => 'select',
+				'options'    => $basket_options,
+				'attributes' => [
+					'data-validation' => 'required',
+				],
+			]
+		);
+
+		$scope_options = [
+			'not set'  => __( '- Select Scope -', 'planet4-master-theme-backend' ),
+			'Global'   => 'Global',
+			'National' => 'National',
+		];
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Scope', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'scope',
+				'type'       => 'select',
+				'options'    => $scope_options,
+				'attributes' => [
+					'data-validation' => 'required',
+				],
+			]
+		);
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Department', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'department',
+				'type'       => 'text_medium',
+				'attributes' => [
+					'placeholder' => __( 'Add Department', 'planet4-master-theme-backend' ),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Populate an associative array with all the children of the ACT page
+	 *
+	 * @return array
+	 */
+	public function populate_act_page_children_options() {
+		$parent_act_id = planet4_get_option( 'act_page' );
+		$options       = [];
+
+		if ( 0 !== absint( $parent_act_id ) ) {
+			$take_action_pages_args = [
+				'post_type'        => 'page',
+				'post_parent'      => $parent_act_id,
+				'post_status'      => 'publish',
+				'orderby'          => 'menu_order',
+				'order'            => 'ASC',
+				'suppress_filters' => false,
+				'numberposts'      => -1,
```

Comment:
could you please use here a big enough integer in order to follow WP coding standards?